### DISCUSSION TITLE
Access "Box select" and other events from plotly express

### DISCUSSION THREAD

**Discussion by aljosahafner** [zauberzeug/nicegui#1818] [Ideas / Feature Requests]
### Question

Hey,

First of all, I have been using nicegui for quite a while with really nice results. Thanks a lot for the great work you are doing here.
However, recently I have stumbled upon a problem for which I haven't been able to find a solution. Within my application I have a viewer of 2D arrays using plotly.express' imshow function (don't forget to use ```binary_string=True``` for large arrays!). Now I'd like to access the "box select" feature of plotly and use the selection in my application. I didn't find where in the dict, the selected points can be accessed.

![image](https://github.com/zauberzeug/nicegui/assets/51963146/20f7ab0d-ba5e-4399-ac39-90fe81f23b0c)

Could anyone help me?

Thanks

**Comment by falkoschindler** [comment]
Hi, @aljosahafner!
I just played around with something like
```py
p = ui.plotly(go.Figure(go.Scatter(x=[1, 2, 3, 4], y=[1, 2, 3, 2.5])))
p.on('plotlySelected', lambda event: print(event))
```
but it seems like we can't register to Plotly events at the moment. We'll probably need to forward them in plotly.vue, but I'm unsure how to do that generically without specifying every single event individually.

Anyway, I'll convert your question into a feature request. It would definitely be a valuable extension for the `ui.plotly` element.

**Reply by falkoschindler** [reply]
By the way: I don't think the selected points are stored in Plotly, but only sent through the "selected" event. Therefore I think we need to get this event. But maybe I'm wrong and there's a different solution for getting the selected points?

**Comment by aljosahafner** [comment]
@falkoschindler First of all, thanks a lot for replying and taking a look at the matter!

There are two main pages from plotly documentation that are (sort of) describing this. Maybe this helps:
https://plotly.com/python/v3/selection-events/
https://dash.plotly.com/interactive-graphing

What you posted above would be perfect, either registering it as an event or having the box selection points (x_i, y_i) extractable. I am trying to implement an auto-contrast feature similar to ImageJ, where you select a certain area from a 2D image and then click "auto" and the image colorbar is adjusted according to the min and max values within the box.

**Comment by falkoschindler** [comment]
I'll close this feature request as resolved in PR #2231.
